DBContextApp/Repository/GenericRepository.cs

```diff
@@ -0,0 +1,52 @@
+﻿using DBContextApp.Context;
+using Microsoft.EntityFrameworkCore;
+
+namespace DBContextApp.Repository
+{
+    public class GenericRepository : BaseRepository
```

Comment:
Why do you need both generic repo and base repo? You could have one with virtual methods.